lib/active_record/connection_adapters/sqlserver/database_tasks.rb

```diff
@@ -20,11 +20,11 @@ def current_database
         end
 
         def charset
-          select_value "SELECT DATABASEPROPERTYEX(DB_NAME(), 'SqlCharSetName')"
+          select_value "SELECT CAST(DATABASEPROPERTYEX(DB_NAME(), 'SqlCharSetName') AS NVARCHAR(128))"
```

Comment:
Not sure why this needs to be done. Isn't this statement execute in isolation all the time and never with count?